src/main/webapp/3Dscript.js

```diff
@@ -168,19 +234,91 @@ function plotOrientation() {
     matrix.multiply(new THREE.Matrix4().makeRotationY(
         THREE.Math.degToRad(point.rollDeg)));
     matrix.multiply(new THREE.Matrix4().makeTranslation(0.0, .05, 0.0));
-    orientation.setMatrixAt(i, matrix);
+    orientation.setMatrixAt(pose.indexOf(point), matrix);
   }
   orientation.instanceMatrix.needsUpdate = true;
   orientation.position.x = poseTransform.translateX;
   orientation.position.z = poseTransform.translateZ;
   orientation.rotation.y = THREE.Math.degToRad(poseTransform.rotate);
 }
 
+/**
+* get time in readable units
+* @param {String} time raw time up to 6 decimal places. from GPS time
+* @return {String} readable time.
+ */
+function formatTime(time) {
+  time += '';
+  const x = time.split('.');
+  let x1 = x[0];
+  const x2 = x.length > 1 ? '.' + x[1] : '';
+  const rgx = /(\d+)(\d{3})/;
+  while (rgx.test(x1)) {
+    x1 = x1.replace(rgx, '$1' + ',' + '$2');
+  }
+  return x1 + x2;
+}
+
+/**
+* Does matrix multiplication on specific index of the direction matrix.
+* @param {object} typeOfMatrix matrix to multiply by.
+* @param {int} index of point to change.
+*/
+function updateMatrixAtIndex(typeOfMatrix, index) {
```

Comment:
I think I understand this now, but it can still be clearer. Consider something like this:
```
function multiplyInstanceMatrix(mesh, index, multiplicand) {
  let instance;
  mesh.getMatrixAt(index, instance);
  instance.multiply(multiplicand);
  mesh.setMatrixAt(index, instance);
  mesh.instanceMatrix.needsUpdate = true;
}
```
* The name explicitly describes the purpose ("update" could mean any number of things).
* The function gets all of its data as arguments, so the reader doesn't have to track down global variables in order to understand it (you would pass `orientation` as the first argument when calling it). You might even be able to reuse it for a different `InstancedMesh` later.
* `multiplicand` is more accurate than `typeOfMatrix` (which suggests the variable isn't actually a matrix, but some type or classification). It's not my favorite variable name, but it at least suggests that the argument will be on the right side of the multiplication (though I don't see a lot of people using "multiplier * multiplicand = product" for matrices). You could also choose `post` or just `m` or something, with appropriate comments.